packages/app/src/client/utils/grafana.ts

```diff
@@ -46,7 +47,7 @@ export async function createFolder({
   integration,
   tenant
 }: FolderProps): Promise<FolderInfo> {
-  const responseData = await axios({
+  const responseData = await request({
```

Comment:
can you add a quick statement about why we think that the timeout and retrying parameters are a good choice for this request?

Reply:
Both requests are usually part of a larger "request cascade". E.g. [when uninstalling integrations](https://github.com/opstrace/opstrace/blob/ac1a8ee15120656fa7e6e416a0ff29949b03ef18/packages/app/src/client/integrations/common/UninstallIntegrationBtn.tsx#L57-L61). 
As those are not atomic transactions, it would be unfortunate if only some of the requests go through and others time out due to temp network issues. 

Though I would love to hear your input on this, as I am not 100% sure if this is actually the way to go.

Reply:
things we need to understand:

- expected request processing time on the API handler. Less than one second? Less than 60 seconds? What's a reasonable upper bound beyond which we can safely assume thing is wrong?
- idempotency: are these safe to retry (ok when applied multiple times)?

Reply:
> if only some of the requests go through and others time out due to temp network issues.

Oh, ouchie. That kind of state should of course ideally be managed in the backend to make this kind of pseudo-atomic :).

But yeah I get you now that we're here and have to manage this transaction in the UI we have to try real hard to make all the requests succeed.

> the timeout and retrying parameters

With that I really meant the retry count, retry strategy, and timeout constants (the specific numbers)

Reply:
For clarity and for posterity: the retrying parameterization for the `request` object landed here: https://github.com/opstrace/opstrace/pull/1188/files#diff-f0b50639134328642967d005943bc837e05ff06b143b5c7dc56edaef8b4975cbR22 -- global timeout 5 seconds, 3 retries. Sounds good for all those HTTP requests that are expected to be handled _blazingly fast_.

For some other context recently I did set rather custom retrying parameters: https://github.com/opstrace/opstrace/blob/606fa4810cdec52ff6c1314b7c7ff4151144214b/packages/app/src/client/components/withSession/index.tsx#L528